src/shim.c

```diff
@@ -703,6 +704,9 @@ handle_proxy_message(struct cc_shim *shim)
 
 	fr = read_frame(shim);
 	if ( !fr) {
+		if (! reconnect_to_proxy(shim)) {
```

Comment:
If I understand you well, you want to change the `return false` with `abort()` in case you get header length or header version which is wrong ? Because you consider those cases like real issues and we should just terminate the shim, right ?